### PR TITLE
Added action to generate a changelog from Jenkins

### PR DESCRIPTION
**Pull request by mandrizzle**
I run automated internal builds for our iOS and Android apps and found it really useful to generate a changelog using Jenkins. Other changelog generators require a "from" commit to generate a log up to the current HEAD. But this is done automatically with Jenkins. Every job in Jenkins generates a changelog from the HEAD of the last build. This means there is no need to keep track of anything. Just add the action in your lane and let Jenkins do all the work

Thoughts?

### CODE REVIEW

File: lib/fastlane/actions/make_changelog_from_jenkins.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      FL_CHANGELOG = :FL_CHANGELOG
+    end
+
+    class MakeChangelogFromJenkinsAction < Action
+      def self.run(params)
+        require 'nokogiri'
```
Comment (KrauseFx) [inline lib/fastlane/actions/make_changelog_from_jenkins.rb]: We don't want `nokogiri` as a fastlane dependency. Could you solve this problem with some built-in libraries instead?

Comment (KrauseFx) [inline lib/fastlane/actions/make_changelog_from_jenkins.rb]: If not, you could always add 
```ruby
Actions.verify_gem!('nokogiri')
```
before calling `require 'nokogiri'`

**Comment by KrauseFx** [comment]
Looks good, thanks for the PR, however I couldn't test it on a real system. Please update the PR according to my comments and squash the commits :+1:

**Comment by mandrizzle** [comment]
Is there a built in xml parser I can use?

File: lib/fastlane/actions/make_changelog_from_jenkins.rb (hunk 1 above)
Comment (mandrizzle) [inline lib/fastlane/actions/make_changelog_from_jenkins.rb]: Is there a built in xml parser I can use?

**Comment by mandrizzle** [comment]
@KrauseFx Should I have this in my action:

```ruby
module SharedValues
      FL_CHANGELOG = :FL_CHANGELOG
end
```
The compiler gives a warning that it already is defined

**Comment by KrauseFx** [comment]
@mandrizzle Yeah, you should remove the `SharedValues` module

**Comment by mandrizzle** [comment]
Removed the `SharedValues` module

**Comment by KrauseFx** [comment]
Looks great, thanks for your contribution @mandrizzle :+1: This will be in the next release :+1:

**Pull request merged**